impl/src/rendering/templates/partials/crates_macro.template

```diff
@@ -0,0 +1,183 @@
+# A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+DEPENDENCIES = {
+{%- for workspace_member in workspace.workspace_members %}
+{%- if workspace_member %}
+    "{{ bazel_package_name }}/{{ workspace_member }}": {
+{%- else %}
+    "{{ bazel_package_name }}": {
+{%- endif %}
+{%- for crate in crates %}
+{%- if not crate.is_proc_macro and workspace_member in crate.workspace_member_dependents %}
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
+        "{{ crate.pkg_name }}": "{{ crate.workspace_path_to_crate }}:{{ crate_name_sanitized }}",
+{%- endif %}
+{%- endfor %}
+    },
+{%- endfor %}
+}
+
+# A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+PROC_MACRO_DEPENDENCIES = {
+{%- for workspace_member in workspace.workspace_members %}
+{%- if workspace_member %}
+    "{{ bazel_package_name }}/{{ workspace_member }}": {
+{%- else %}
+    "{{ bazel_package_name }}": {
+{%- endif %}
+{%- for crate in crates %}
+{%- if crate.is_proc_macro and workspace_member in crate.workspace_member_dependents %}
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
+        "{{ crate.pkg_name }}": "{{ crate.workspace_path_to_crate }}:{{ crate_name_sanitized }}",
+{%- endif %}
+{%- endfor %}
+    },
+{%- endfor %}
+}
+
+# A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+DEV_DEPENDENCIES = {
+{%- for workspace_member in workspace.workspace_members %}
+{%- if workspace_member %}
+    "{{ bazel_package_name }}/{{ workspace_member }}": {
+{%- else %}
+    "{{ bazel_package_name }}": {
+{%- endif %}
+{%- for crate in crates %}
+{%- if not crate.is_proc_macro and workspace_member in crate.workspace_member_dev_dependents %}
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
+        "{{ crate.pkg_name }}": "{{ crate.workspace_path_to_crate }}:{{ crate_name_sanitized }}",
+{%- endif %}
+{%- endfor %}
+    },
+{%- endfor %}
+}
+
+# A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+DEV_PROC_MACRO_DEPENDENCIES = {
+{%- for workspace_member in workspace.workspace_members %}
+{%- if workspace_member %}
+    "{{ bazel_package_name }}/{{ workspace_member }}": {
+{%- else %}
+    "{{ bazel_package_name }}": {
+{%- endif %}
+{%- for crate in crates %}
+{%- if crate.is_proc_macro and workspace_member in crate.workspace_member_dev_dependents %}
+{%- set crate_name_sanitized = crate.pkg_name | replace(from="-", to="_") %}
+        "{{ crate.pkg_name }}": "{{ crate.workspace_path_to_crate }}:{{ crate_name_sanitized }}",
+{%- endif %}
+{%- endfor %}
+    },
+{%- endfor %}
+}
+
+def crates(deps):
```

Comment:
What do you think about splitting this out into a separate `bzl` file which is not a template, and is just copied during generation? That way it can be edited and linted with regular tools.

I recognize that it depends on the definitions of the maps. These could be brought in via `load` I think, with a hard coded path local to the file itself, like this:

```bzl
load(":crates.bzl",
     "DEPENDENCIES", 
     "PROC_MACRO_DEPENDENCIES", 
     "DEV_DEPENDENCIES", 
     "DEV_PROC_MACRO_DEPENDENCIES")
```

Reply:
`crates.bzl` is buildifier compliant (proven by the passing CI) or maybe I misunderstand what you mean.

Can you also elaborate on what you're trying to accomplish by loading each map individually? I almost made them private but didn't want to be restrictive. Users should only ever interact with the `crates` and `all_crates` macros.

Reply:
My request here is for you to separate the non-dynamic parts of this file from the generated parts of this file. I think this is easier to understand for future maintainers because the crate maps are simple generated maps, and the functions are just functions in a bzl file. Developers will be able to easily and automatically run buildifier on the bzl file. This approach also eliminates the temptation of doing code generation inside of the bzl functions, which I think is a perilous temptation which would inevitably result in a large increase in complexity.

Reply:
I think the design has changed to not require this. Gonna resolve it but definitely reopen it if you feel otherwise.